packages/react-native-platform-override/src/Cli.ts

```diff
@@ -314,56 +323,61 @@ function printValidationErrors(errors: Array<ValidationError>) {
   // Add an initial line of separation
   console.error();
 
-  errors.sort((a, b) => a.file.localeCompare(b.file));
-
-  const filesMissing = errors.filter(err => {
-    return err.type === 'fileMissingFromManifest';
-  });
-  const overridesMissing = errors.filter(err => {
-    return err.type === 'overrideFileNotFound';
-  });
-  const baseFilesNotFound = errors.filter(err => {
-    return err.type === 'baseFileNotFound';
-  });
-  const outOfDateFiles = errors.filter(err => {
-    return err.type === 'outOfDate';
-  });
-
-  if (filesMissing.length > 0) {
-    const errorMessage = `Found override files that aren't listed in the manifest. Overrides can be added to the manifest by using 'npx ${
+  printErrorType(
+    'missingFromManifest',
+    errors,
+    `Found override files that aren't listed in the manifest. Overrides can be added to the manifest by using 'npx ${
       npmPackage.name
-    } add <override>':`;
-    console.error(chalk.red(errorMessage));
-    filesMissing.forEach(err => console.error(` - ${err.file}`));
-    console.error();
-  }
+    } add <override>':`,
+  );
 
-  if (overridesMissing.length > 0) {
-    const errorMessage = `Found overrides in the manifest that don't exist on disk. Remove existing overrides using 'npx ${
+  printErrorType(
+    'overrideNotFound',
+    errors,
+    `Found overrides in the manifest that don't exist on disk. Remove existing overrides using 'npx ${
       npmPackage.name
-    } remove <override>':`;
-    console.error(chalk.red(errorMessage));
-    overridesMissing.forEach(err => console.error(` - ${err.file}`));
-    console.error();
-  }
+    } remove <override>':`,
+  );
 
-  if (baseFilesNotFound.length > 0) {
-    const errorMessage = `Found overrides whose original files do not exist. Remove existing overrides using 'npx ${
+  printErrorType(
+    'baseNotFound',
+    errors,
+    `Found overrides whose base files do not exist. Remove existing overrides using 'npx ${
       npmPackage.name
-    } remove <override>':`;
-    console.error(chalk.red(errorMessage));
-    baseFilesNotFound.forEach(err => console.error(` - ${err.file}`));
-    console.error();
-  }
+    } remove <override>':`,
+  );
 
-  if (outOfDateFiles.length > 0) {
-    const errorMessage = `Found overrides whose original files have changed. Upgrade overrides using 'npx ${
+  printErrorType(
+    'outOfDate',
+    errors,
+    `Found overrides whose original files have changed. Upgrade overrides using 'npx ${
       npmPackage.name
     } auto-upgrade <manifest>' and 'npx ${
       npmPackage.name
-    } manual-upgrade <manifest>':`;
-    console.error(chalk.red(errorMessage));
-    outOfDateFiles.forEach(err => console.error(` - ${err.file}`));
+    } manual-upgrade <manifest>':`,
+  );
+
+  printErrorType(
+    'overrideDifferentFromBase',
+    errors,
+    'The following overrides should be an exact copy of their base files. Ensure overrides are up to date or revert changes:',
+  );
+}
```

Comment:
future suggestion: maybe a catch-all here? 
i.e. have printErrorType remove the errors printed from the errors list and then here cehck if there are any errors left?

Reply:
Addressed in the next iteration.